src/App.js

```diff
@@ -12,29 +17,23 @@ function App() {
 	const [cart, setCart] = useState([]);
 
 	const addItem = item => {
-		// add the given item to the cart
+		setCart([...cart, item]);
 	};
 
 	return (
 		<div className="App">
-			<Navigation cart={cart} />
+		<CartContext.Provider value={cart}>
+			<ProductContext.Provider value={{ products, addItem }}>
+				<Navigation cart={cart} />
```

Comment:
Because we are using context, we can remove the `cart={cart}` when we render the `<Navigation/>` component '' '